include/swift/AST/NameLookup.h

```diff
@@ -27,21 +27,38 @@
 #include "swift/Basic/SourceLoc.h"
 #include "swift/Basic/SourceManager.h"
 #include "llvm/ADT/SmallVector.h"
+#include "swift/AST/TypeRepr.h"
 
 namespace swift {
 class ASTContext;
+class ASTWalker;
 class DeclName;
 class Type;
 class TypeDecl;
 class ValueDecl;
 struct SelfBounds;
 class NominalTypeDecl;
+class Evaluator;
 
 namespace ast_scope {
 class ASTSourceFileScope;
 class ASTScopeImpl;
 } // namespace ast_scope
 
+using CollectedOpaqueReprs = SmallVector<TypeRepr *, 2>;
+
+/// Walk the type representation recursively, collecting any
+/// `OpaqueReturnTypeRepr`s.
+CollectedOpaqueReprs collectOpaqueReturnTypeReprs(TypeRepr *);
+
+/// Walk the type representation recursively, collecting any
+/// `CompositionTypeRepr`s and `IdentTypeRepr`s.
+CollectedOpaqueReprs collectTypeReprs(TypeRepr *);
```

Comment:
This name is too general since it doesn't collect _all_ type representations, only the protocol (and protocol composition presumably). How about collectProtocolTypeReprs instead?

Reply:
I renamed this to `collectOpaqueReturnTypeReprs` and removed the second function in my latest commit. 
Is it too soon to just call this `collectOpaqueTypeReprs` ? 👀